docs: IA (P0): add Work the queue "Route and assign suggestions to the right people" page (#731)
Promptless/docs: src/content/docs/docs/work-the-queue/assignment-and-routing.mdx

Reviewer: We need an acknowledgement that usernames are different across platforms and that Promptless uses a best-effort approach to identify users across platforms. But to help Promptless along, you can add a username map in your Agent Knowledge Base to map usernames of common users across platforms.

---
title: Route and assign suggestions to the right people
description: >-
  Learn how Promptless routes notifications, assigns documentation PRs, and
  credits requesters so the right person picks up each suggestion
slug: docs/work-the-queue/assignment-and-routing
sidebar:
  hidden: false
  order: 29
  label: Assignment & routing
---
import { Aside } from '@astrojs/starlight/components';

> A suggestion only helps if it reaches someone who can review it. Promptless routes each suggestion—and the pull request it opens—to the people closest to the change: it notifies whoever requested the work, assigns the documentation PR, and credits the original requester in the PR. You shape that routing through your [Agent Knowledge Base](/docs/get-the-most-out/agent-knowledge-base), so it follows your team's ownership and review conventions rather than a one-size-fits-all default.

Reviewer: ```suggestion
A suggestion only helps if it reaches someone who can review it. Promptless routes each suggestion, or the pull request it opens, to the people closest to the change: it notifies whoever requested the work, assigns the documentation PR, and credits the original requester in the PR. You shape that routing through your [Agent Knowledge Base](/docs/get-the-most-out/agent-knowledge-base), so it follows your team's ownership and review conventions rather than a one-size-fits-all default.
```

<Aside type="note">

> This page covers how work is routed to people—notifications, assignees, and requester credit. To choose the Slack or Microsoft Teams channel a notification lands in, see [Customize notifications](/docs/tune/notifications).

Reviewer: ```suggestion
This page covers how work is routed to people, including notifications, assignees, and requesters. To choose the Slack or Microsoft Teams channel a notification lands in, see [Customize notifications](/docs/tune/notifications).
```

</Aside>

## Notify the person who requested the change

> When Promptless announces a suggestion, it @mentions the person the work came from—the author of the merged pull request, the teammate who tagged Promptless in Slack, or whoever started the thread. The requester sees the suggestion in the channel and knows it's waiting on them, so the notification never lands without an owner.

Reviewer: ```suggestion
When Promptless announces a suggestion, it notifies the person the work came from, whether that's the author of the merged pull request, the teammate who tagged Promptless in Slack, or whoever started the thread. The requester sees the suggestion in the channel and knows it's waiting on them, so the notification never lands without an owner.
```

> Because Promptless ties each notification back to its source, it can also reply where the conversation already lives. When a suggestion comes from a Slack thread, Promptless can post the update as a reply in that thread, so everyone already discussing the change sees it in context.

Reviewer: ```suggestion
Because Promptless ties each notification back to its source, it can also reply where the conversation already lives. When a suggestion comes from a Slack thread, Promptless can post the update as a reply in that thread, so everyone already discussing the change sees it in context, while a source PR in GitHub might get a comment about a new related docs PR.
```

## Assign the documentation PR

> When Promptless opens a documentation PR, it sets an assignee so the PR lands in that person's review queue. Out of the box, with no configuration, it assigns the person the change came from—the source PR's author, or the teammate who requested the update in Slack—matching their account across GitHub and Slack.

Reviewer: @Promptless, is this grounded in actual code behavior? Can you cite it?

Assigning the PR to the requester keeps ownership continuous: whoever changed the code or asked for the update is the same person GitHub prompts to review the docs that follow.

<Aside type="note">
Assigned a Promptless docs PR and want to know what to do with it? See [Review and edit Promptless PRs](/docs/work-the-queue/reviewing-prs) to review the diff, request changes, or steer it from a GitHub comment.
</Aside>

## Credit the requester in the PR description

For a suggestion that starts from a Slack request, Promptless names the requester on a dedicated line near the top of the PR description, formatted as `Requested by: <Full Name>`. This shows at a glance who asked for the change, so a reviewer picking up the PR later knows who to follow up with.

## Route approvals to a reviewer

> Routing doesn't stop when the PR opens. When a documentation PR is approved but your designated writer or reviewer hasn't reviewed it yet, Promptless can hand the PR off to them—setting them as an assignee and sending a notification that the PR is ready for a final writer review. Content-focused reviewers get a clear signal that a PR has cleared its first approval and now needs their eyes, and no one has to reassign it by hand.

Reviewer: This requires customization in the Agent Knowledge Base.

Usernames rarely line up across platforms—the same person might be `@jdoe` in Slack and `jane-doe` on GitHub. Promptless matches accounts on a best-effort basis, so it usually connects a requester's identities on its own. When it can't, or when you want to be sure, add a username map to your [Agent Knowledge Base](/docs/get-the-most-out/agent-knowledge-base) that pairs each teammate's Slack and GitHub handles. Promptless then uses that map to assign and credit the right person even when their names differ.

<Aside type="note">
  Assigned a Promptless docs PR and want to know what to do with it? See [Review and edit Promptless PRs](/docs/work-the-queue/reviewing-prs) to review the diff, request changes, or steer it from a GitHub comment.
</Aside>

## Credit the requester in the PR description

For a suggestion that starts from a Slack request, Promptless names the requester on a dedicated line near the top of the PR description, formatted as `Requested by: <Full Name>`. This shows at a glance who asked for the change, so a reviewer picking up the PR later knows who to follow up with.

## Route approvals to a reviewer

> Routing doesn't stop when the PR opens. You can have Promptless hand an approved PR to a designated writer or reviewer. When a documentation PR receives an approval and that writer or reviewer hasn't approved it themselves, Promptless sets them as an assignee and notifies them that the PR is ready for writer review. You set this handoff up in your Agent Knowledge Base, so it follows the writer or reviewer you name rather than running by default. That writer or reviewer gets a clear signal that a PR has cleared an approval and now needs their eyes, and no one has to reassign it by hand.

Reviewer: ```suggestion
Routing doesn't stop when the PR opens. You can have Promptless hand an approved PR to a designated writer or reviewer. When a documentation PR receives an approval and that writer or reviewer hasn't approved it themselves, Promptless sets them as an assignee and notifies them that the PR is ready for writer review. You set this handoff up in your Agent Knowledge Base, so it follows the writer or reviewer you name rather than running by default. That writer or reviewer gets a clear signal that a PR has cleared an approval and now needs their eyes, and no one has to reassign it by hand.

<Aside type="note">
  If you want to support this workflow, make sure your GitHub PR triggers are set to activate when a PR gets it's first approval.
</Aside>
```

## Customize routing in your Knowledge Base

> Your [Agent Knowledge Base](https://app.gopromptless.ai/settings/knowledge) governs routing and assignment, so you can adapt it to how your team divides ownership. Organization admins can edit these preferences directly; other members have read-only access.

Reviewer: ```suggestion
Your [Agent Knowledge Base](https://app.gopromptless.ai/settings/knowledge) governs routing and assignment, so you can adapt it to how your team divides ownership. Organization admins can edit these preferences directly. Other members have read-only access.
```

You write these preferences in plain language—not a config schema—and Promptless applies them. Common adjustments include:

- **Who gets notified**: @mention the original requester, or route suggestions to an area owner based on which docs or product surface the change touches.
- **Who owns the PR**: choose which teammate is assigned a documentation PR.
- **How approvals hand off**: name the writer or reviewer a PR should route to once it's approved, and how they're notified.
- **Who's who across platforms**: map a teammate's Slack and GitHub handles so Promptless assigns and credits the right person when their usernames differ.

For example, you might add lines like:

```
When a PR is approved but our lead writer hasn't approved it, assign them and notify them for a final review.
Route suggestions that touch the billing docs to the billing team's owner.

> Jane Doe is @jdoe in Slack and jane-doe on GitHub.

Reviewer: ```suggestion
When a PR is approved but our lead writer (Jane) hasn't approved it, assign them and notify them for a final review.
Route suggestions that touch the billing docs to the billing team's owner (Juan).

| Person | Slack | GitHub |
| --- | --- | --- |
| Jane | @jdoe | jane-doe |
| Bill | @Bill | billhandler |
```

```

> Notification and routing preferences live in the `doc_workflow/notification_preferences.md` file within your Knowledge Base. It's a lightweight scaffold for your overrides—Promptless handles the underlying templates and delivery. For the full range of notification customizations and channel configuration, see [Customize notifications](/docs/tune/notifications). To teach Promptless durable conventions from your review feedback, see [Teach conventions with feedback](/docs/tune/teaching-conventions).

Reviewer: ```suggestion
Notification and routing preferences live in the `doc_workflow/notification_preferences.md` file within your Knowledge Base. It's a lightweight scaffold for your overrides, while Promptless handles the underlying templates and delivery. For the full range of notification customizations and channel configuration, see [Customize notifications](/docs/tune/notifications). To teach Promptless durable conventions from your review feedback, see [Teach conventions with feedback](/docs/tune/teaching-conventions).
```

> Need help routing suggestions to the right people? Contact us at [help@gopromptless.ai](mailto\:help@gopromptless.ai).

Reviewer: ```suggestion
```
